src/tasks/migration/utils.ts

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import * as fse from "fs-extra";
+import * as path from "path";
+import { debounce } from "lodash";
+import { tasks, window, workspace } from "vscode";
+import { DeprecatedExportJarTaskProvider, ExportJarTaskProvider } from "../../exportJarSteps/ExportJarTaskProvider";
+import { contextManager } from "../../contextManager";
+import { Context } from "../../constants";
+import { DiagnosticProvider } from "./DiagnosticProvider";
+import { CodeActionProvider } from "./CodeActionProvider";
+
+export async function getTasksJsonPaths(): Promise<string[]> {
+    const tasksJsonPaths = [];
+    if (workspace.workspaceFolders?.length) {
+        for (const folder of workspace.workspaceFolders) {
+            const tasksJsonPath = path.join(folder.uri.fsPath, ".vscode", "tasks.json");
+            if (await fse.pathExists(tasksJsonPath)) {
+                tasksJsonPaths.push(tasksJsonPath);
+            }
+        }
+    }
+    return tasksJsonPaths;
+}
+
+export function asyncDebounce(func: any, wait: any, bind: any) {
+    const debounced = debounce(async (resolve, reject, bindSelf, args) => {
+        try {
+            const result = await func.bind(bindSelf)(...args);
+            resolve(result);
+        } catch (error) {
+            reject(error);
+        }
+    }, wait);
+
+    function returnFunc(...args: any[]) {
+        return new Promise((resolve, reject) => {
+            debounced(resolve, reject, bind, args);
+        });
+    }
+
+    return returnFunc;
+}
+
+export async function updateExportTaskType(tasksJsonPath: string): Promise<void> {
+    const content = await fse.readFile(tasksJsonPath);
+    const taskJSON = JSON.parse(content.toString());
+    if (taskJSON) {
+        const taskArray = taskJSON.tasks;
+        if (taskArray instanceof Array) {
+            for (const task of taskArray) {
+                if (task.type === DeprecatedExportJarTaskProvider.type) {
+                    task.type = ExportJarTaskProvider.exportJarType;
+                }
+            }
+        }
+        await fse.writeFile(tasksJsonPath, JSON.stringify(taskJSON, null, "\t"));
+    }
+}
+
+export async function handleDeprecatedTasks(): Promise<void> {
```

Comment:
Is this function being used anywhere?

Reply:
we check the status in `activateExtension()` when activating. it will pop up a notification if deprecated tasks detected.

Reply:
Oh yes, I missed it.

Reply:
I now have a feeling that, pops up such notifications is not a good UX. I think we should:

- Provide option like `Don't ask again`
- Only show it when user opens a `tasks.json` file

**Or maybe even just don't pop it, since we have diagnostics.**

It just like when we deprecate a setting, VS Code will only show diagnostics and will not pop notifications for them.